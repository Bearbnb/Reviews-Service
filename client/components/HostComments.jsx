import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import styles from '../../styles/Review.css';

const moment = require('moment');

const HostComments = ({ review, host }) => {
  return (
      <div className={styles.hostContainer}>
    <div className={styles.mainHostContainer}>
      <div className={styles.hostPhoto}>
        <div className={styles.hostAvatar}>
          <img className={styles.avatar} src={host.photo} alt="" />
        </div>
      </div>
      <div className={styles.hostText}>
        <div className={styles.textHostHeader}>
          <h4 className={styles.response}>Response from {host.name}:</h4>
        </div>
          <p className={styles.response}>{review.host_comments}</p>
        <div className={styles.hostText} />
        <div className={styles.HostDate}>
          {moment(review.created).format('MMMM YYYY')}
        </div>
      </div>
    </div>
  </div>
  );
};

HostComments.propTypes = {
  review: PropTypes.object.isRequired,
  host: PropTypes.object.isRequired,
};

export default HostComments;
