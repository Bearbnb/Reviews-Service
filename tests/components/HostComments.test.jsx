import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import HostComments from '../../client/components/HostComments';

describe('<HostComments />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<HostComments review={{
      host: { photo: '', host_comments: '' },
      review: {},
    }}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
