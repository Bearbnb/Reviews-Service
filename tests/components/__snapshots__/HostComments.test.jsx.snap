// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HostComments /> should render properly 1`] = `
<div
  className="hostContainer"
>
  <div
    className="mainHostContainer"
  >
    <div
      className="hostPhoto"
    >
      <div
        className="hostAvatar"
      >
        <img
          alt=""
          className="avatar"
          src=""
        />
      </div>
    </div>
    <div
      className="hostText"
    >
      <div
        className="textHostHeader"
      >
        <h4
          className="response"
        >
          Response from 
          :
        </h4>
      </div>
      <p
        className="response"
      />
      <div
        className="hostText"
      />
      <div
        className="HostDate"
      >
        September 2018
      </div>
    </div>
  </div>
</div>
`;
